components/web/firefox/patches/webrtc-solaris.patch

```diff
@@ -0,0 +1,19393 @@
+--- firefox-60.8.0/media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build
++++ firefox-patched/media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build
+@@ -161,6 +161,19 @@ if CONFIG["OS_TARGET"] == "OpenBSD":
+     DEFINES["WEBRTC_POSIX"] = True
+     DEFINES["_FILE_OFFSET_BITS"] = "64"
+ 
++if CONFIG["OS_TARGET"] == "SunOS":
++
++    DEFINES["USE_X11"] = "1"
++    DEFINES["WEBRTC_POSIX"] = True
++    DEFINES["WEBRTC_SOLARIS"] = True
++    DEFINES["_FILE_OFFSET_BITS"] = "64"
++
++    UNIFIED_SOURCES += [
++        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
++        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
++        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
++    ]
++
+ if CONFIG["OS_TARGET"] == "WINNT":
+ 
+     DEFINES["CERT_CHAIN_PARA_HAS_EXTRA_FIELDS"] = True
+@@ -216,6 +229,10 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+ 
+     DEFINES["_FORTIFY_SOURCE"] = "2"
+ 
++if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "SunOS":
++
++    DEFINES["_FORTIFY_SOURCE"] = "2"
++
+ if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
+ 
+     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
+diff --git firefox-60.8.0/media/webrtc/trunk/webrtc/common_audio/common_audio_gn/moz.build firefox-patched/media/webrtc/trunk/webrtc/common_audio/common_audio_gn/moz.build
+index b93db3279..c4e2a0f84 100644
+--- firefox-60.8.0/media/webrtc/trunk/webrtc/common_audio/common_audio_gn/moz.build
++++ firefox-patched/media/webrtc/trunk/webrtc/common_audio/common_audio_gn/moz.build
+@@ -129,6 +129,13 @@ if CONFIG["OS_TARGET"] == "OpenBSD":
+     DEFINES["WEBRTC_POSIX"] = True
+     DEFINES["_FILE_OFFSET_BITS"] = "64"
+ 
++if CONFIG["OS_TARGET"] == "SunOS":
++
++    DEFINES["USE_X11"] = "1"
++    DEFINES["WEBRTC_POSIX"] = True
++    DEFINES["WEBRTC_SOLARIS"] = True
++    DEFINES["_FILE_OFFSET_BITS"] = "64"
++
+ if CONFIG["OS_TARGET"] == "WINNT":
+ 
+     DEFINES["CERT_CHAIN_PARA_HAS_EXTRA_FIELDS"] = True
+@@ -182,6 +189,10 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+ 
+     DEFINES["_FORTIFY_SOURCE"] = "2"
+ 
++if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "SunOS":
++
++    DEFINES["_FORTIFY_SOURCE"] = "2"
++
+ if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
+ 
+     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
+diff --git firefox-60.8.0/media/webrtc/trunk/webrtc/common_audio/common_audio_sse2_gn/moz.build firefox-patched/media/webrtc/trunk/webrtc/common_audio/common_audio_sse2_gn/moz.build
+index 763c439e0..ec2331733 100644
+--- firefox-60.8.0/media/webrtc/trunk/webrtc/common_audio/common_audio_sse2_gn/moz.build
++++ firefox-patched/media/webrtc/trunk/webrtc/common_audio/common_audio_sse2_gn/moz.build
+@@ -111,6 +111,7 @@ if CONFIG["OS_TARGET"] == "Linux":
+ if CONFIG["OS_TARGET"] == "NetBSD":
+ 
+     CXXFLAGS += [
++        "-msse2",
```

Comment:
Why does this necessary?

Reply:
don't need to run GN or GYP to regenerate the Chromium build system at compile time, unless you would prefer this?

Reply:
I mean changes to NetBSD-specific CXXFLAGS

Reply:
ah, must have been added by GN itself, as I had not touched any of the other platform flags. I'll take a look again...